packages/core/src/amazonq/lsp/lspClient.ts

```diff
@@ -0,0 +1,219 @@
+        process.env.Q_WORKER_THREADS = workerThreads.toString()
+    } else {
+        delete process.env.Q_WORKER_THREADS
+    }
+
+    const nodename = process.platform === 'win32' ? 'node.exe' : 'node'
+
+    const child = cp.spawn(extensionContext.asAbsolutePath(path.join('resources', nodename)), [
+        serverModule,
+        ...debugOptions.execArgv,
+    ])
+    // share an encryption key using stdin
+    // follow same practice of DEXP LSP server
+    writeEncryptionInit(child.stdin)
+
+    // If the extension is launch in debug mode the debug server options are use
+    // Otherwise the run options are used
+    let serverOptions: ServerOptions = {
+        run: { module: serverModule, transport: TransportKind.ipc },
+        debug: { module: serverModule, transport: TransportKind.ipc, options: debugOptions },
+    }
+
+    serverOptions = () => Promise.resolve(child!)
+
+    const documentSelector = [{ scheme: 'file', language: '*' }]
+
+    // Options to control the language client
+    const clientOptions: LanguageClientOptions = {
+        // Register the server for json documents
+        documentSelector,
+        initializationOptions: {
+            handledSchemaProtocols: ['file', 'untitled'], // language server only loads file-URI. Fetching schemas with other protocols ('http'...) are made on the client.
+            provideFormatter: false, // tell the server to not provide formatting capability and ignore the `aws.stepfunctions.asl.format.enable` setting.
+            extensionPath: extensionContext.extensionPath,
+        },
+    }
+
+    // Create the language client and start the client.
+    LspClient.instance.client = new LanguageClient(
+        'codewhisperer',
+        localize('codewhisperer.server.name', 'Amazon Q Language Server'),
+        serverOptions,
+        clientOptions
+    )
+    LspClient.instance.client.registerProposedFeatures()
+
+    const disposable = LspClient.instance.client.start()
+    toDispose.push(disposable)
+
+    let savedDocument: vscode.Uri | undefined = undefined
+
+    toDispose.push(
+        vscode.workspace.onDidSaveTextDocument(document => {
+            if (document.uri.scheme !== 'file') {
+                return
+            }
+            savedDocument = document.uri
+        })
+    )
+    toDispose.push(
+        vscode.window.onDidChangeActiveTextEditor(editor => {
+            if (savedDocument && editor && editor.document.uri.fsPath !== savedDocument.fsPath) {
+                void LspClient.instance.updateIndex(savedDocument.fsPath)
+            }
+        })
+    )
+    return LspClient.instance.client.onReady().then(() => {
+        const disposableFunc = { dispose: () => rangeFormatting?.dispose() as void }
+        toDispose.push(disposableFunc)
+    })
+}
+
+export async function deactivate(): Promise<any> {
+    if (!LspClient.instance.client) {
+        return undefined
+    }
+    return LspClient.instance.client.stop()
+}
```

Comment:
is this func being used? i didn't see other reference

Reply:
1. There are strange dependency issue if I add this in deactivate of extension
2. Without calling this function, lsp process is still shutdown if I quit VS Code.